Automattic/woocommerce-payments
# Capture authorizations from table

Fixes #

#### Changes proposed in this Pull Request
This PR adds a Capture button to each row of the new authorizations list table. This button will allow merchants to capture an authorization from the `Uncaptured` tab in the `Transactions` page in WP Admin. 

It uses mocked data, since the server endpoints are not available yet. For this reason, it is not yet possible to manually test the complete flow. Only the loading and error states can be tested manually at this moment.

The scope of changes on this PR includes:

- UI components: adds a new `CaptureAuthorizationButton` to the list of non captured authorizations
- State management boilerplate in the front end
- Refactor `capture_terminal_payment` endpoint. The functionality needed to capture a terminal payment and to capture an authorization are very similar. To avoid code duplication, I refactored the endpoint to make it work for both types of payments, passing a new `$is_terminal_payment` parameter.


https://user-images.githubusercontent.com/1553182/188485604-d5de638e-8af2-491e-964b-8416a8ac8a3b.mp4











#### Testing instructions





* Set `displayAuthorizations = true ` [here](https://github.com/Automattic/woocommerce-payments/blob/e9fec5a7a492ecbe59171991867e4f831b9d1296/client/transactions/index.tsx#L24). Don't forget to rebuild statics (i.e. `npm start`)
* Navigate to `WP Admin > Transactions > Uncaptured` tab. You should see a `Capture` button in each row of the table.
* Click on any `Capture` button
* Check that the button transitions to loading state
* Check that it errors eventually and displays an error notice.

-------------------


**Post merge**
